### PR TITLE
Add eva plotting to soca vrfy task

### LINKED ISSUE #202: Generate marine DA plots using EVA in the GDASApp
- [ ] Modify the existing python tool in the GDASApp [gen_eva_obs_yaml.py](https://github.com/NOAA-EMC/GDASApp/blob/develop/ush/eva/gen_eva_obs_yaml.py) to generate obs space plots using a valid ```var.yaml``` from marine DA experiments
- [ ] create eva templates for the marine obs
- [ ] test the generated yaml with eva

### LINKED ISSUE #348: Miss-guiding comment in vrfy j-job
Per discussion with @guillaumevernieres, `ocnanalpost` is the dependency of both `ocnanalvrfy` and the following cycle, due to the anticipated run length of `ocnanalvrfy`. `JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY` for its part has this comment

`# Do not remove the Temporary working directory (do this in POST)`

and `JGDAS_GLOBAL_OCEAN_ANALYSIS_POST` does indeed have the default to remove `DATA`. The apparent assumption in   `exgdas_global_marine_analysis_vrfy.py` is to run in `COMOUT`, but with a todo to run in `DATA`, which I suspect is what is expected in global-workflow. As such there appears to be the at least the future danger of `ocnanalpost` deleting `DATA` before `ocnanalvrfy` is finished in it.

### LINKED ISSUE #362: ioda change in diag file layout might break eva yaml generator
Apparently a recent change in ioda removes the processor indicator from diag file filenames, eg from `adt_j3.2018041512_0000.nc4` to `adt_j3.2018041512.nc4`, which potentially breaks the yamls generated by`ush/eva/gen_eva_obs_yaml.py` handle filenames. At the very least it does on the marine side, as I discovered in final testing of additions for using eva to generate plots for soca. Would changing this in `ush/eva/gen_eva_obs_yaml.py` work for everybody else?

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
Adds eva plotting to vrfy task(https://github.com/NOAA-EMC/GDASApp/issues/202), fixes bug in eva yaml generator (https://github.com/NOAA-EMC/GDASApp/issues/362), and cleans up comments (https://github.com/NOAA-EMC/GDASApp/issues/348)

Files changed:
scripts/exgdas_global_marine_analysis_vrfy.py
ush/eva/gen_eva_obs_yaml.py

- fixes #202 
- fixes #362 
- fixes #348

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
First try for the automated `CI` ... ppfff, you're just showing off @AndrewEichmann-NOAA .

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: this is not going to be available if you are not in CMake land...

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: lol what?

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: isn't that py script 'importable'? if so, why not just call the main function within?

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: Same comment.

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_vrfy.py:227]: Same comment.

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_vrfy.py:217]: I'm not proud of the subprocess thing - we should probably make the yaml generator a function, and *I* should have made the marine yaml script one

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: my bad

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: this would have been superfluous anyway - turns out I'd caught it already

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: I'd forgotten I'd made this into a function

File: scripts/exgdas_global_marine_analysis_vrfy.py
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_vrfy.py:227]: This one seems to be unavoidable?

**Comment by AndrewEichmann-NOAA** [comment]
I've addressed the subprocess calls as much as I think I can, but this requires a change to the jjob which first must be merged before merging this, but I want to get this approved before submitting the g-w PR

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Blocking this PR until the j-job is updated

**Comment by AndrewEichmann-NOAA** [comment]
restarted CI test on Hera now that https://github.com/NOAA-EMC/global-workflow/pull/1331 is merged

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Pull request merged**
